polaris-react/src/components/Image/Image.tsx

```diff
@@ -1,38 +1,66 @@
-import React from 'react';
+import React, {useCallback, useEffect} from 'react';
+
+import {classNames} from '../../utilities/css';
+
+import styles from './Image.scss';
 
 interface SourceSet {
   source: string;
   descriptor?: string;
 }
 
 type CrossOrigin = 'anonymous' | 'use-credentials' | '' | undefined;
+type Status = 'loading' | 'loaded';
 
 export interface ImageProps extends React.HTMLProps<HTMLImageElement> {
   alt: string;
   source: string;
   crossOrigin?: CrossOrigin;
   sourceSet?: SourceSet[];
+  noPlaceholder?: boolean;
```

Comment:
Maybe seeing a ton of gray blocks on an image heavy page isn't the right thing to do. Offering an escape hatch

Reply:
Can we not add the escape hatch and see if this is a problem?

Reply:
Knew you'd say that 😄